### PR TITLE
fix: issue #6

### PR DESCRIPTION
**Pull request by lkwinta**
Fix issue #6

### CODE REVIEW

**Comment by lubiluk** [comment]
Why do you even consider clamping. You comment says the model can output invalid actions during inference. However, in the discrete mode it is trained with cross entropy, so it's impossible to output actions from outside of the dictionary. Perhaps you had some other issue, with padding episodes maybe?

**Comment by lkwinta** [comment]
I have done some cleaning with act() in agent.py, I will provide some more comments in code for the changes later on

**Comment by lkwinta** [comment]
> Why do you even consider clamping. You comment says the model can output invalid actions during inference. However, in the discrete mode it is trained with cross entropy, so it's impossible to output actions from outside of the dictionary. Perhaps you had some other issue, with padding episodes maybe?

the comment is unfortunate, I meant issue with embedding or one hot encoding later on wihin this function because we set act dim=1 and vocab_size=4 but we had 5 tokens for embedding

maybe infering vocab_size like this  `self.model_config.act_vocab_size = dataset.action_space.n + 1` will be the solusion to this

File: src/nanodt/agent.py [hunk 1]
```diff
@@ -197,7 +211,7 @@ def act(self, obs, rew=None):
             states, max_len, pad_value=0, device=device, dtype=torch.float32
         )
         actions = pad_tensor(
-            actions, max_len, pad_value=-1, device=device, dtype=torch.long
+            actions, max_len, pad_value=0, device=device, dtype=torch.float32
```
Comment (lubiluk) [inline src/nanodt/agent.py]: Please revert it to -1, to not change semantics and comment near clamp() why we do clamping.

**Review by lubiluk — CHANGES_REQUESTED** [review]
These are important fixes, thank you. There is one inconsistency introduced here: during training we pad with -1, since this value is treated specially and ignored in the loss computation. I would avoid changing this semantics for inference as well.

Therefore, please keep padding with -1 and retain the clamping logic in inference. Just add an appropriate comment explaining this consistency. Once that is updated, I’ll merge.

File: src/nanodt/agent.py (hunk 1 above)
Comment (lkwinta) [reply src/nanodt/agent.py]: Done
